File: src/C3.Catalogue/Decks/Deck.vb
Namespace Decks

    Public NotInheritable Class Deck

        Public Sub New(name As String, addedAt As DateTime, details As DeckDetails)
            If details Is Nothing Then
                Throw New ArgumentNullException("details")
            End If
            Me.Name = name
            Me.AddedAt = addedAt
            Me.Details = details
        End Sub

        Public ReadOnly Property Name As String
        Public ReadOnly Property AddedAt As DateTime
        Public ReadOnly Property Details As DeckDetails

    End Class

End Namespace

File: src/C3.Catalogue/Decks/Deck.cs
using System;

namespace C3.Catalogue.Decks
{
    public sealed class Deck
    {
        public Deck(string name, DateTime addedAt, DeckDetails details)
        {
            if (details == null)
            {
                throw new ArgumentNullException(nameof(details));
            }

            Name = name;
            AddedAt = addedAt;
            Details = details;
        }

        public string Name { get; }

        public DateTime AddedAt { get; }

        public DeckDetails Details { get; }
    }
}
